### ISSUE TITLE
Verified layer stuck (Release v0.2.0-rc4)

### ISSUE THREAD

**Issue by avive** [spacemeshos/go-spacemesh#2689]
## Description
Verified layer stuck on devnet [Release v0.2.0-rc4](https://github.com/spacemeshos/go-spacemesh/releases/tag/untagged-f72f6af266eabcd53dcf)

## Steps To Reproduce
1. Start a node with devnet config file and sync - verified layer advances.
2. Stop it and start it again after ~20 minutes.
3. Let it run for ~1 hour.

## Behavior
Verified layer doesn't advance.
```
$ status node
Node info:
Version: v0.2.0-rc4
Build: cc8e72c8959faf0afef3ea8b21bbefef6ae5c260
API server: localhost:9092 (GRPC API 1.1). >> Insecure Connection. Use only with a local trusted server <<
Synced: true
Synced layer: 50
Current layer: 50
Verified layer: 21
Peers: 8
```

## Expected Behavior
Verified layer to advance to tip.

## Resources
Full node logs attached.

[logs.txt.zip](https://github.com/spacemeshos/go-spacemesh/files/7006109/logs.txt.zip)

**Comment by dshulyak** [comment]
I haven't checked logs attached to the issue, but i can reproduce it locally everytime.

```
2021-08-18T16:24:58.833+0300	INFO	1d5eb.trtl         	rescoring all blocks in old layer using weak coin	{"node_id": "1d5eb2f0ce0f5b5fda2387b648cb2c92b3b50fb48b1aca8e7466dc6cc7ea6bce", "module": "trtl", "tortoise_rerun": "false", "layer_id": 12, "candidate_block_id": "080bf862a0", "count": 26, "coinflip": false, "coinflip_layer": 121, "name": "trtl"}
2021-08-18T16:24:58.840+0300	WARN	1d5eb.trtl         	global opinion on block differs from our vote, cannot verify layer	{"node_id": "1d5eb2f0ce0f5b5fda2387b648cb2c92b3b50fb48b1aca8e7466dc6cc7ea6bce", "module": "trtl", "tortoise_rerun": "false", "verification_target": 122, "old_verified": 11, "candidate_layer": 12, "block_id": "68cd1c9ee6", "global_opinion": "abstain (+0, -0)", "local_opinion": "against (+0, -1)", "name": "trtl"}
2021-08-18T16:24:58.840+0300	WARN	1d5eb.trtl         	global opinion on block is abstain, cannot verify layer	{"node_id": "1d5eb2f0ce0f5b5fda2387b648cb2c92b3b50fb48b1aca8e7466dc6cc7ea6bce", "module": "trtl", "tortoise_rerun": "false", "verification_target": 122, "old_verified": 11, "candidate_layer": 12, "block_id": "68cd1c9ee6", "global_opinion": "abstain (+0, -0)", "local_opinion": "against (+0, -1)", "name": "trtl"}
2021-08-18T16:24:58.840+0300	INFO	1d5eb.trtl         	self healing attempting to verify candidate layer	{"node_id": "1d5eb2f0ce0f5b5fda2387b648cb2c92b3b50fb48b1aca8e7466dc6cc7ea6bce", "module": "trtl", "tortoise_rerun": "false", "old_verified_layer": 11, "last_verified_layer": 11, "target_layer": 112, "candidate_layer": 12, "last_layer_received": 122, "hdist": 10}
2021-08-18T16:24:58.847+0300	INFO	1d5eb.trtl         	self healing failed to verify candidate layer, will reattempt later	{"node_id": "1d5eb2f0ce0f5b5fda2387b648cb2c92b3b50fb48b1aca8e7466dc6cc7ea6bce", "module": "trtl", "tortoise_rerun": "false", "old_verified_layer": 11, "last_verified_layer": 11, "target_layer": 112, "candidate_layer": 12, "last_layer_received": 122, "hdist": 10, "candidate_block_id": "080bf862a0", "name": "trtl"}
```

This is the result after everything is synced.

**Comment by avive** [comment]
In my case, verified layer was advancing up to 21 on first node session but got stuck only after a node restart.

**Comment by lrettig** [comment]
This should've been fixed by the recent fixes in #2400, reopen if it recurs
